sbi/utils/get_nn_models.py

```diff
@@ -213,6 +213,12 @@ def posterior_nn(
     )
 
     if model == "mdn_snpe_a":
+        if num_components != 10:
```

Comment:
The default for `SNPE_A.__init__()` is also `num_components=10`. Thus, the algorithm will perform as expected.

Reply:
I see, so when the users provides their own `_build_density_estimator` fun via `posterior_nn`, then the `num_components` basically has no meaning? They would have to leave it with the default and instead pass it when creating the `SNPE_A` object. 
But this is a bit weird, no? 
Alternative suggestion: 
remove the `num_components` from the `__init__` of `SNPE_A` and instead infer `self._num_components` from the default `self._neural_net`. 
In fact, I might be missing sth, but one could just set it to `1` for all but the last round, and otherwise rely on what is the default in `_build_neural_net` or on what was passed by the user the `posterior_nn`.

Reply:
in particular: not use a `partial` but just the default `_build_neural_net` in line `181` and `187`, and use the `partial` as it is done at the moment to set `num_components=1` for the first rounds.

Reply:
for snpe-a, with default behavor, `_build_neural_net` is never called with `num_components`. That's because for the first round, we have only a single component. Thus, when the neural net is built, it is built with a single component. In the last round, we generate an MDN, but **not** by calling `_build_neural_net` but by replicating `num_components`. Lines `181` and `187` are only used if `retrain_from_scratch=True`.

Reply:
Yes, I know, but from the user perspective this is quite confusing. Currently, we have the additional `num_components` `arg` in SNPEA to do the repetition of the single components Gaussian in the last round. But instead we could get rid of the additional arg and just infer `num_components` from the default neural net built in `snpe_base`. Then the user will have only one place to specify the number of components, by passing their own `build_neural_net` fun obtained from `posterior_nn` (as in all other inference methods).

Reply:
OK got, we would not have access to `num_components` because it is just an arg of the `_build_neural_net` `Callable`  that is called only during training in `snpe_base`. 
So I agree, the current way is good, let's just add a bit more information in the docstrings.